### ISSUE TITLE
Using nginx runs php-fpm under the wrong user

### ISSUE THREAD

**Issue by entendu** [geerlingguy/drupal-vm#781]
Using nginx, the worker process runs as user `nginx`, but the php-fpm pools run as user `apache`. This causes issues with things like log files, where (for example) pimpmylog expects logs to be owned by `apache`, when they're owned by `nginx`. Other goofy issues happen too, like `/var/log/php-fpm` being owned by `apache`.

config.yml is fairly standard, set up for a local environment (no makes/builds). Relevant:
```
# The web server software to use. Can be either 'apache' or 'nginx'.
drupalvm_webserver: nginx
```
After ssh'ing into the newly built vm:
```
$ ps aux | grep nginx
root      1214  0.0  0.1  48084  1500 ?        Ss   13:50   0:00 nginx: master process /usr/sbin/nginx -c /etc/nginx/nginx.conf
nginx     1215  0.0  0.2  48880  3008 ?        S    13:50   0:00 nginx: worker process
```
and
```
$ ps aux | grep apache
apache    1260  0.0  1.3 652112 14056 ?        S    13:50   0:00 php-fpm: pool www
apache    1261  0.0  1.2 652088 12328 ?        S    13:50   0:00 php-fpm: pool www
apache    1262  0.0  1.1 652152 11412 ?        S    13:50   0:00 php-fpm: pool www
apache    1263  0.0  1.6 652344 16432 ?        S    13:50   0:00 php-fpm: pool www
apache    1264  0.0  1.2 652152 12760 ?        S    13:50   0:00 php-fpm: pool www
```

I expect this might be a downstream problem with the geerlingguy.nginx role, but I haven't dug in enough to find out yet.

**Comment by entendu** [comment]
Ah, this is actually a part of geerlingguy/ansible-role-php:

```
- name: Define php_fpm_pool_user.
  set_fact:
    php_fpm_pool_user: "{{ __php_fpm_pool_user }}"
  when: php_fpm_pool_user is not defined
```
and `__php_fpm_pool_user` is either `www-data` or `apache` depending on OS. It seems to me that php-fpm's user should depend on the http server's user as a default, no?

**Comment by geerlingguy** [comment]
For now, you can override that user using that variable in your own `config.yml`; but we could look into ways of making it more reliable in all scenarios upstream in the `geerlingguy.php` role.

**Comment by entendu** [comment]
Yep, makes sense. Would it be worth putting a comment in the example config.yml? Out of the box, pimpmylog does not work with in an nginx scenario because of this. Would be happy to whip up a PR for it.

**Comment by entendu** [comment]
Interestingly, I tried doing this in config.yml and it did NOT overwrite the user. Any thoughts or debugging tips?

**Comment by geerlingguy** [comment]
@entendu - Was that after a fresh Drupal VM `vagrant up`? It might've stuck with the old user if your reprovisioned an existing VM. But if that variable doesn't do what it says it's doing, then we should file an issue upstream in the php role.

**Comment by entendu** [comment]
Master of drupal-vm still references php role 3.2.1, which is missing this fix: geerlingguy/ansible-role-php#128. See my PR #794 to bump version & add comments/defaults to default.config.yml.

**Comment by geerlingguy** [comment]
Ah, that's it! Good catch!
